liquigraph-core/src/main/java/org/liquigraph/core/io/ReplaceChecksumWriter.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2014-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.liquigraph.core.io;
+
+import org.liquigraph.core.model.Changeset;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.liquigraph.core.exception.Throwables.propagate;
+
+public class ReplaceChecksumWriter {
+
+    private static final String REPLACE_CHECKSUM_UPDATE =
+        "MATCH (changeset:__LiquigraphChangeset {id: ?, author: ?}) " +
+            "SET  changeset.checksum = ?";
```

Comment:
it should happen for changesets that currently do not have a changeset (it's probably the case as you currently filter the changesets beforehand) but since this query is likely to move to `ChangelogGraphReader`, this query will need to be updated

Reply:
In commit 9754dce: The writer is used as is and called by ReplaceChecksumGraphReader which filters the changesets to update before calling the writer.